lib/geordi/db_cleaner.rb

```diff
@@ -165,17 +171,27 @@ def list_all_dbs(dbtype)
       else
         list_all_mysql_dbs
       end
+    rescue DatabaseError
+      Interaction.fail 'Connection to database could not be established. Try running the command again with --sudo'
     end
 
     def list_all_postgres_dbs
-      `#{@postgres_command} -t -A -c 'SELECT DATNAME FROM pg_database WHERE datistemplate = false'`.split
+      output, error, _status = Open3.capture3("#{@postgres_command} -t -A -c 'SELECT DATNAME FROM pg_database WHERE datistemplate = false'")
+
+      raise DatabaseError unless error.empty?
```

Comment:
Usually, the status is an even better means of recognizing an error. You can even ask `status.success?`. Some commands write to stderr even if overall successful.